File: prototype/Assets/microcosmicWar/Scripts/zz/GUI/zzGUI.js

var myGUI:zzInterfaceGUI;

static function impGUI(pMyGUI:zzInterfaceGUI)
{
	//GUI.depth = myGUI.depth;
	
	if(pMyGUI.getVisible())
	{
		var lSkin:GUISkin = pMyGUI.getSkin();
		if(lSkin)
		{
			//สนำร Skin
			var lPreSkin :GUISkin = GUI.skin ;
			GUI.skin = lSkin;
			pMyGUI.impGUI();
			GUI.skin = lPreSkin;
		}
		else
			pMyGUI.impGUI();
	}
}

function impGUI()
{
	impGUI(myGUI);
}

function getDepth()
{
	return myGUI.depth;
}

function setGUI(pGUI:zzInterfaceGUI)
{
	myGUI= pGUI;
}

function getGUI()
{
	return myGUI;
}

File: prototype/Assets/microcosmicWar/Scripts/zz/GUI/zzGUI.cs

using UnityEngine;
using System.Collections;

public class zzGUI : MonoBehaviour
{


    public zzInterfaceGUI myGUI;

    public static void impGUI(zzInterfaceGUI pMyGUI)
    {
        //GUI.depth = myGUI.depth;

        if (pMyGUI.getVisible())
        {
            GUISkin lSkin = pMyGUI.getSkin();
            if (lSkin)
            {
                //使用 Skin
                GUISkin lPreSkin = GUI.skin;
                GUI.skin = lSkin;
                pMyGUI.impGUI();
                GUI.skin = lPreSkin;
            }
            else
                pMyGUI.impGUI();
        }
    }

    public void impGUI()
    {
        impGUI(myGUI);
    }

    public int getDepth()
    {
        return myGUI.depth;
    }

    public void setGUI(zzInterfaceGUI pGUI)
    {
        myGUI = pGUI;
    }

    public zzInterfaceGUI getGUI()
    {
        return myGUI;
    }
}